Funcky/Extensions/QueryableExtensions/FirstOrNone.cs

```diff
@@ -16,7 +16,7 @@ public static partial class QueryableExtensions
         public static Option<TSource> FirstOrNone<TSource>(this IQueryable<TSource> source)
             where TSource : notnull
             => source
-                .Select(x => Option.Some(x))
+                .Select(Option.Some)
```

Comment:
That lambda is intentionally there. You're now using the extension for `IEnumerable` which is not good.

Reply:
Why are all tests green then? :)

Reply:
I see the problem, maybe a comment?

Reply:
I tried to add tests here: https://github.com/polyadic/funcky/pull/401